tests/test_scheduler.py

```diff
@@ -94,7 +98,11 @@ def check_over_configs(self, time_step=0, **config):
             elif num_inference_steps is not None and not hasattr(scheduler, "set_timesteps"):
                 kwargs["num_inference_steps"] = num_inference_steps
 
+            # Set the seed before state as some schedulers are stochastic like EulerAncestralDiscreteScheduler, EulerDiscreteScheduler
+            torch.manual_seed(0)
             output = scheduler.step(residual, time_step, sample, **kwargs).prev_sample
+
+            torch.manual_seed(0)
```

Comment:
setting seed before each `step` as the euler schedulers are stochastic.

Reply:
Even better would be to pass a `generator` to step IMO :-)